File: 26-Remove-Duplicates-from-Sorted-Array/main.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        index = 1
        val = nums[0]
        for i in nums[1:]:
            if val == i:
                continue
            else:
                val = nums[index] = i
                index += 1
        return index

    def output(self, nums, size):
        print(nums[:size])

if __name__ == '__main__':
    nums = [1, 2, 2, 3, 3]
    sln = Solution()

    size = sln.removeDuplicates(nums)
    sln.output(nums, size)

File: 26-Remove-Duplicates-from-Sorted-Array/main.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates01(vector<int>& nums) {
        if (nums.size() <= 1) {
            return nums.size();
        }
        int index = 0;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[index] != nums[i]) {
                nums[++index] = nums[i];
            }
        }
        return index + 1;
    }

    void testcase(vector<int>& nums) {
        int arr[4] = {1, 1, 2, 2};
        vector<int> vec(arr, arr + sizeof(arr) / sizeof(int));
        nums.swap(vec);
    }

    void output(const vector<int>& nums, int size) {
       for (int i = 0; i < size; i++) {
           cout << nums[i] << " ";
       }
       cout << endl << "size: " << size << endl;
    }
}; 

int main() {
   vector<int> vec;
   int size;
   
   Solution* sln = new Solution();

   sln->testcase(vec);
   size = sln->removeDuplicates01(vec);
   sln->output(vec, size);
}
